tests/analysis_tests/tests/src/DotPipeCompleteFromCurrentModule.res

```diff
@@ -14,4 +14,81 @@ module Y = {
     }
 
     let b = (x:t) => 4
+}
+
+
+module Types = {
+    type comp
+    type context
+    type vec2
+}
+
+module PosComp = (
+  T: {
+    type t
+  },
+) => {
+  open Types
+
+  @send
+  external addPos: (context, float, float) => comp = "pos"
+
+  @send
+  external addPosFromVec2: (context, vec2) => comp = "pos"
+}
+
+module SpriteComp = (
+    T: {
+        type t
+    }
+)
+ => {
+    open Types
+
+    @send
+    external addSprite: (context, string) => comp = "sprite"
+ }
+
+external k: Types.context = "k"
+
+@send
+external add: (Types.context, array<Types.comp>) => 't = "add"
+
+module Wall = {
+    type t
+
+    include PosComp({ type t = t })
+
+    let blah = (k: Types.context) => ""
+
+    let make = () => {
+        [
+            // k.
+            //   ^com
+
+            // add
+            //    ^com
+        ]
+    }
+
+    let makeWith = (x) => {
+        k->add([
+            k->addPos(1.0, 2.0),
+            // addP
+            //     ^com
+        ])
+    }
+
+    module Poster = {
+        type t
+
+        include SpriteComp({ type t = t })
+
+        let make = () => {
+            [
+                // k.
+                //   ^com
+            ]
+        }
+    }
```

Comment:
One distinction between open and include is that only include supports the following:

```res
module M = {
  let x = 3
}

module N = { include M }

let q = N.x
```

I think it already works, completing a module from the outside, so it should just be a matter of adding to the tests.

Reply:
Can that be something for a follow up PR? It is outside of my scope, and thus matters less to me. The PR right now would be a huge win to have.

Reply:
I think it already works (on functors), just needs to extend the test a bit.